src/mapeo-project.js

```diff
@@ -749,6 +761,389 @@ export class MapeoProject extends TypedEmitter {
+   */
+  async #zipFileName(observations, tracks) {
+    let exportType = ''
+    if (observations) exportType += 'Obsvns'
+    if (tracks) {
+      if (observations) exportType += '_'
+      exportType += 'Tracks'
+    }
+    const prefix = await this.#exportPrefix(exportType)
+
+    return prefix + '.zip'
+  }
+
+  /**
+   * Export observations and or tracks as a GeoJSON file
+   * @param {string} exportFolder Path to save the file. The file name is auto-generated
+   * @param {Object} [options={}]
+   * @param {boolean} [options.observations=true] Whether observations should be exported
+   * @param {boolean} [options.tracks=true] Whether all tracks and their observations should be exported
+   * @param {string} [options.lang]
+   * @returns {Promise<string>} The full path that the file was exported at
+   */
+  async exportGeoJSONFile(
+    exportFolder,
+    { observations = true, tracks = true, lang } = {}
+  ) {
+    const fileName = await this[kGeoJSONFileName](observations, tracks)
+    const filePath = path.join(exportFolder, fileName)
+    const source = this[kExportGeoJSONStream]({ observations, tracks, lang })
+    const sink = createWriteStream(filePath)
+    await pipelinePromise(source, sink)
+
+    return filePath
+  }
+
+  /**
+   * @param {Attachment} attachment
+   * @returns {Promise<null | Readable<Uint8Array|Buffer>>}
+   */
+  async #tryExportBlob(attachment) {
+    // Audio must not have variants
+    for (const variant of VARIANT_EXPORT_ORDER) {
+      const blobId = buildBlobId(attachment, variant)
+      const entry = await this.#blobStore.entry(blobId)
+      if (!entry) continue
+      // Download
+      // @ts-expect-error
+      return this.#blobStore.createReadStream(blobId)
+    }
+
+    return null
+  }
+
+  /**
+   * @param {ZipArchive} archive
+   * @param {Object} [options={}]
+   * @param {boolean} [options.observations=true] Whether observations should be exported
+   * @param {boolean} [options.tracks=true] Whether all tracks and their observations should be exported
+   * @param {boolean} [options.attachments=true] Whether all attachments for observations should be exported
+   * @param {string} [options.lang]
+   * @returns {Promise<void>}
+   */
+  async #exportToArchive(
+    archive,
+    { observations = true, tracks = true, attachments = true, lang } = {}
+  ) {
+    // GeoJSON
+    const geoJSONFileName = await this[kGeoJSONFileName](observations, tracks)
+    const seenAttachments = new Map()
+    const geoJSONStream = this[kExportGeoJSONStream]({
+      observations,
+      tracks,
+      lang,
+      seenAttachments,
+    })
+
+    // @ts-expect-error
+    await archive.entry(geoJSONStream, { name: geoJSONFileName })
+
+    const missingAttachments = []
+    // Attachments
+    if (attachments) {
+      const mediaFolder = this.#exportPrefix('Media') + '/'
+      for (const attachment of seenAttachments.values()) {
+        const stream = await this.#tryExportBlob(attachment)
```

Comment:
for clarity, the output for an exported attachment is a singular blob associated with the variant of preference, where the order of that preference is original > preview > thumbnail?

wondering if we'll need to eventually export all variants of an attachment at some point, which accounts for situations where the blob types used between variants might be notably different for some reason (example provided in https://github.com/digidem/comapeo-core/issues/903#issuecomment-2428707408). probably introduces a lot of complexity in terms of implementation but seeing this reminded me of that scenario.

Reply:
IMO the API surface can be extended to allow for more options once we get a requirement for it. So far the spec says this on attachment export:

> When including attachments, make a “best effort” - try for original variant first, and fallback to preview, then thumbnail, if not available.

One thing this made me notice though is I wasn't adding the variant suffix! I'll make the fix.